src/main/java/net/fabricmc/loader/launch/FabricTweaker.java

```diff
@@ -88,13 +101,19 @@ public void injectIntoClassLoader(LaunchClassLoader launchClassLoader) {
 
 		launchClassLoader.addClassLoaderExclusion("net.fabricmc.api.Environment");
 		launchClassLoader.addClassLoaderExclusion("net.fabricmc.api.EnvType");
+		launchClassLoader.addClassLoaderExclusion("net.fabricmc.api.ModInitializer");
```

Comment:
Hmm is there a reason specifically why we need to exclude the mod initializers?

Reply:
Yes, it's explained in the commit message:

> These interfaces are accessed from the Entrypoint* classes which themselves are
> excluded from the LaunchClassLoader. Not excluding the interfaces will cause
> those to be loaded by the LaunchClassLoader as well as its parent, leading to
> casting errors when mod entry points are to be invoked.

https://github.com/FabricMC/fabric-loader/pull/288/commits/49f6f49173fa59af07c24558e9031df58b7b6d2b

Reply:
I see. I've let modmuss and player know about this PR so they can look into it since they have a better grasp on loader internals than I do